### PR TITLE
Skip distribution if refund fails to send on channel closure

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #1396
---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/29-fee/keeper/escrow.go [hunk 1]
```diff
@@ -214,21 +215,16 @@ func (k Keeper) RefundFeesOnChannelClosure(ctx sdk.Context, portID, channelID st
 				return err
 			}
 
-			// if the refund address is blocked, skip and continue distribution
-			if k.bankKeeper.BlockedAddr(refundAddr) {
-				continue
-			}
```
Comment (chatton) [inline modules/apps/29-fee/keeper/escrow.go:220]: this blocked check happens in `SendCoinsFromModuleToAccount` now instead

File: modules/apps/29-fee/keeper/escrow.go [hunk 2]
```diff
@@ -214,21 +215,16 @@ func (k Keeper) RefundFeesOnChannelClosure(ctx sdk.Context, portID, channelID st
...
-				return err
+			if err = k.bankKeeper.SendCoinsFromModuleToAccount(cacheCtx, types.ModuleName, refundAddr, packetFee.Fee.Total()); err != nil {
+				failedToSendCoins = true
+				continue
 			}
-
 		}
 
-		k.DeleteFeesInEscrow(cacheCtx, identifiedPacketFee.PacketId)
+		if !failedToSendCoins {
```
Comment (chatton) [inline modules/apps/29-fee/keeper/escrow.go:225]: it's possible an error occurred sending the coins, we don't want to remove them unless they were actually sent.

File: modules/apps/29-fee/keeper/escrow.go [hunk 3]
```diff
@@ -214,21 +215,16 @@ func (k Keeper) RefundFeesOnChannelClosure(ctx sdk.Context, portID, channelID st
 				return err
```
Comment (charleenfei) [inline modules/apps/29-fee/keeper/escrow.go:215]: i think this error also still needs to be handled so improperly formatted refund addr doesnt stop channel closure?

Comment (chatton) [reply modules/apps/29-fee/keeper/escrow.go:215]: ah that's a good point, it's effectively the same situation. Nice catch!

Comment (charleenfei) [reply modules/apps/29-fee/keeper/escrow.go:215]: 🤝

**Review by charleenfei — APPROVED** [review]
lgtm! 🎉

File: modules/apps/29-fee/keeper/escrow_test.go [hunk 4]
```diff
@@ -2,11 +2,10 @@ package keeper_test
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	"github.com/tendermint/tendermint/crypto/secp256k1"
-
 	"github.com/cosmos/ibc-go/v3/modules/apps/29-fee/types"
 	transfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
 	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
```
Comment (damiannolan) [inline modules/apps/29-fee/keeper/escrow_test.go:8]: nit: group imports by type, see #382 

```go
import (
        // standard library imports
	"fmt"
	"testing"
        
        // external library imports
	"github.com/stretchr/testify/require"
	abci "github.com/tendermint/tendermint/abci/types"
        
         // ibc-go library imports
	"github.com/cosmos/ibc-go/modules/core/23-commitment/types"
)
```

**Pull request merged**